Flaky-test: PendingTopicFuturesTest#testExceptionalComplete

PendingTopicFuturesTest is flaky. The testExceptionalComplete test method fails sporadically.

Add `invocationCount = 100` and test it in my local env. There is one time it failed.

```
18:40:00.342 [TestNG-method=testExceptionalComplete-1:io.streamnative.pulsar.handlers.kop.PendingTopicFuturesTest@113] INFO  io.streamnative.pulsar.handlers.kop.PendingTopicFuturesTest - exceptionMessages: [error, error, error, error, error, error, error, error, error, error]
18:40:00.343 [TestNG-method=testExceptionalComplete-1:io.streamnative.pulsar.handlers.kop.PendingTopicFuturesTest@117] INFO  io.streamnative.pulsar.handlers.kop.PendingTopicFuturesTest - Changes of pending futures' count: [1, 2, 3, 4, 1, 0, 0, 0, 0, 0]


java.lang.AssertionError: Lists differ at element [4]: 5 != 1 expected [5] but found [1]
Expected :5
Actual   :1
<Click to see difference>


	at org.testng.Assert.fail(Assert.java:96)
	at org.testng.Assert.failNotEquals(Assert.java:776)
	at org.testng.Assert.assertEqualsImpl(Assert.java:137)
	at org.testng.Assert.assertEquals(Assert.java:831)
	at org.testng.Assert.assertEquals(Assert.java:796)
	at io.streamnative.pulsar.handlers.kop.PendingTopicFuturesTest.testExceptionalComplete(PendingTopicFuturesTest.java:120)
	at sun.reflect.GeneratedMethodAccessor1.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:54)
	at org.testng.internal.InvokeMethodRunnable.run(InvokeMethodRunnable.java:44)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

--- a/kafka-impl/src/test/java/io/streamnative/pulsar/handlers/kop/PendingTopicFuturesTest.java
+++ b/kafka-impl/src/test/java/io/streamnative/pulsar/handlers/kop/PendingTopicFuturesTest.java
@@ -18,6 +18,7 @@ import java.util.List;
 import java.util.Optional;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ThreadLocalRandom;
 import lombok.extern.slf4j.Slf4j;
 import org.apache.pulsar.broker.service.persistent.PersistentTopic;
 import org.testng.Assert;
@@ -55,19 +56,19 @@ public class PendingTopicFuturesTest {
     @Test(timeOut = 10000)
     void testNormalComplete() throws ExecutionException, InterruptedException {
         final PendingTopicFutures pendingTopicFutures = new PendingTopicFutures(null);
-        final CompletableFuture<Optional<PersistentTopic>> topicFuture = CompletableFuture.supplyAsync(() -> {
-            sleep(800);
-            return Optional.empty();
-        });
+        final CompletableFuture<Optional<PersistentTopic>> topicFuture = new CompletableFuture<>();
         final List<Integer> completedIndexes = new ArrayList<>();
         final List<Integer> changesOfPendingCount = new ArrayList<>();
+        int randomNum = ThreadLocalRandom.current().nextInt(0, 9);
 
         for (int i = 0; i < 10; i++) {
             final int index = i;
             pendingTopicFutures.addListener(
                     topicFuture, ignored -> completedIndexes.add(index), (ignore) -> {});
             changesOfPendingCount.add(pendingTopicFutures.size());
-            sleep(234);
+            if (randomNum == i) {
+                topicFuture.complete(Optional.empty());
+            }
         }
 
         // all futures are completed, the size becomes 0 again.
@@ -88,19 +89,20 @@ public class PendingTopicFuturesTest {
     @Test(timeOut = 10000)
     void testExceptionalComplete() throws ExecutionException, InterruptedException {
         final PendingTopicFutures pendingTopicFutures = new PendingTopicFutures(null);
-        final CompletableFuture<Optional<PersistentTopic>> topicFuture = CompletableFuture.supplyAsync(() -> {
-            sleep(800);
-            throw new RuntimeException("error");
-        });
+        final CompletableFuture<Optional<PersistentTopic>> topicFuture = new CompletableFuture<>();
         final List<String> exceptionMessages = new ArrayList<>();
         final List<Integer> changesOfPendingCount = new ArrayList<>();
 
+        int randomNum = ThreadLocalRandom.current().nextInt(0, 9);
         for (int i = 0; i < 10; i++) {
             pendingTopicFutures.addListener(topicFuture, topic -> {}, (ex) -> {
                 exceptionMessages.add(ex.getMessage());
             });
             changesOfPendingCount.add(pendingTopicFutures.size());
-            sleep(200);
+
+            if (randomNum == i) {
+                topicFuture.completeExceptionally(new RuntimeException("error"));
+            }
         }
 
         // all futures are completed, the size becomes 0 again.

